### PR TITLE
Add code editor font name as a new user preference

### LINKED ISSUE #4996: [SUGGESTION] Let us choose code editor font
I understand Source Code Pro is pretty popular on Linux systems, but here on Windows there's a bunch of coders that would rather use Consolas instead. So, let us set the font we want to use in the code editor. No need of fancy dialogs, just a text field like the one we use to set external editor will suffice.

Thanks.

### PR DESCRIPTION
**Pull request by britzl**
Fixes #4996

### CODE REVIEW

File: editor/src/clj/editor/code/data.clj [hunk 1]
```diff
@@ -501,7 +501,7 @@
 
 (defn tab-stops [glyph-metrics tab-spaces]
   (let [^double space-width (char-width glyph-metrics \space)
-        tab-width (* space-width (double tab-spaces))]
+        tab-width (max 1 (* space-width (double tab-spaces)))]
```
Comment (britzl) [inline editor/src/clj/editor/code/data.clj:504]: Not sure why, but sometimes the width was reported as 0 which resulted in division by 0 where the tab-width was used

File: editor/src/clj/editor/prefs_dialog.clj [hunk 2]
```diff
@@ -95,14 +95,15 @@
             :prefs [{:label "Enable Texture Compression" :type :boolean :key "general-enable-texture-compression" :default false}
                     {:label "Escape Quits Game" :type :boolean :key "general-quit-on-esc" :default false}
                     {:label "Track Active Tab in Asset Browser" :type :boolean :key "asset-browser-track-active-tab?" :default false}
-                    {:label "Path to custom keymap" :type :string :key "custom-keymap-path" :default ""}]}
+                    {:label "Path to custom keymap" :type :string :key "custom-keymap-path" :default ""}
+                    {:label "Code editor font (requires restart)" :type :string :key "code-editor-font-name" :default "Dejavu Sans Mono"}]}
```
Comment (britzl) [inline editor/src/clj/editor/prefs_dialog.clj:99]: There's probably a clever way to make the code editor view recreate itself when a user preference is changed, but I do not know how.

File: editor/src/clj/editor/code/view.clj [hunk 3]
```diff
@@ -1783,9 +1784,12 @@
   (init-property-and-bind-preference! find-case-sensitive-property prefs "code-editor-find-case-sensitive" false)
   (init-property-and-bind-preference! find-wrap-property prefs "code-editor-find-wrap" true)
   (init-property-and-bind-preference! font-size-property prefs "code-editor-font-size" default-font-size)
+  (init-property-and-bind-preference! font-name-property prefs "code-editor-font-name" "")
   (init-property-and-bind-preference! visible-indentation-guides-property prefs "code-editor-visible-indentation-guides" true)
   (init-property-and-bind-preference! visible-minimap-property prefs "code-editor-visible-minimap" true)
-  (init-property-and-bind-preference! visible-whitespace-property prefs "code-editor-visible-whitespace" true))
+  (init-property-and-bind-preference! visible-whitespace-property prefs "code-editor-visible-whitespace" true)
+  (when (clojure.string/blank? (.getValue font-name-property))
```
Comment (britzl) [inline editor/src/clj/editor/code/view.clj:1791]: If the user has set a blank string in the prefs_dialog we revert back to "Dejavu Sans Mono"

Comment (JCash) [reply editor/src/clj/editor/code/view.clj:1791]: It is a single text field?
Would it be possible to select a font file instead? It feels like it's very easy to misspell the font name?

**Review by JCash — APPROVED** [review]

File: editor/src/clj/editor/code/view.clj (hunk 3 above)
Comment (britzl) [reply editor/src/clj/editor/code/view.clj:1791]: There is no file selector functionality in the Preference dialog unfortunately. It was the same when we got a PR for additional Keymaps. I'll merge it as it is and perhaps we can add that later.

**Pull request merged**